mlflow/pyspark/optuna/storage.py

```diff
@@ -0,0 +1,414 @@
+        key: str
+            Attribute key.
+        value: JSONSerializable
+            Attribute value. It should be JSON serializable.
+        """
+        self._mlflow_client.set_tag(study_id, f"user_{key}", json.dumps(value))
+
+    def set_study_system_attr(self, study_id, key: str, value: JSONSerializable) -> None:
+        """Register a optuna-internal attribute as mlflow run tags to a study run.
+        This method overwrites any existing attribute.
+        Parameters
+        ----------
+        study_id: str
+            mlflow run ID of the study.
+        key: str
+            Attribute key.
+        value: JSONSerializable
+            Attribute value. It should be JSON serializable.
+        """
+        self._mlflow_client.set_tag(study_id, f"sys_{key}", json.dumps(value))
+
+    def get_study_id_from_name(self, study_name: str) -> int:
+        runs = self._search_run_by_name(study_name)
+        if len(runs):
+            return runs[0].info.run_id
+        else:
+            raise Exception(f"Study {study_name} not found")
+
+    def get_study_name_from_id(self, study_id) -> str:
+        run = self._mlflow_client.get_run(study_id)
+        return run.data.tags["mlflow.runName"]
+
+    def get_study_directions(self, study_id) -> List[StudyDirection]:
+        run = self._mlflow_client.get_run(study_id)
+        directions_str = run.data.tags["optuna.study_direction"]
+        directions = [StudyDirection[name] for name in directions_str.split(",")]
+        return directions
+
+    def get_study_user_attrs(self, study_id) -> Dict[str, Any]:
+        run = self._mlflow_client.get_run(study_id)
+        user_attrs = {}
+        for key, value in run.data.tags.items():
+            if key.startswith("user_"):
+                user_attrs[key[5:]] = json.loads(value)
+        return user_attrs
+
+    def get_study_system_attrs(self, study_id) -> Dict[str, Any]:
+        run = self._mlflow_client.get_run(study_id)
+        system_attrs = {}
+        for key, value in run.data.tags.items():
+            if key.startswith("sys_"):
+                system_attrs[key[4:]] = json.loads(value)
+        return system_attrs
+
+    def get_all_studies(self) -> List[FrozenStudy]:
+        # print("Getting all studies")
+        runs = self._mlflow_client.search_runs(experiment_ids=[self._experiment_id])
+        studies = []
+        for run in runs:
+            study_id = run.info.run_id
+            study_name = run.data.tags["mlflow.runName"]
+            directions_str = run.data.tags["optuna.study_direction"]
+            directions = [StudyDirection[name] for name in directions_str.split(",")]
+            studies.append(
+                FrozenStudy(
+                    study_name=study_name,
+                    direction=None,
+                    directions=directions,
+                    user_attrs=self.get_study_user_attrs(study_id),
+                    system_attrs=self.get_study_system_attrs(study_id),
```

Comment:
Do we need to make client calls here? The data in `runs` should be of type `Run` https://github.com/mlflow/mlflow/blob/8ed7d131243ec7a210a87eb8a328024eb93bc2ae/mlflow/entities/run.py and should have all of the params, metrics, tags, metadata, etc.
Can we just search over the runs data to access the information from the object's data structure?